sdk/src/main/java/io/customer/sdk/CustomerIOClient.kt

```diff
@@ -47,7 +48,18 @@ internal class CustomerIOClient(
             }
         }
 
-        val queueStatus = backgroundQueue.addTask(QueueTaskType.IdentifyProfile, IdentifyProfileQueueTaskData(identifier, attributes))
+        // If SDK previously identified profile X and X is being identified again, no use blocking the queue with a queue group.
+        val queueGroupStart = if (isFirstTimeIdentifying || isChangingIdentifiedProfile) QueueTaskGroup.IdentifyProfile(identifier) else null
+        // If there was a previously identified profile, or, we are just adding attributes to an existing profile, we need to block
```

Comment:
Thank you for adding very useful comments, helps a lot in understanding the approach. We talked about this before as well but since these tasks were going to be smaller we said, it might not be worth moving tasks as separate classes/entities. 

Coming back to that, I feel like this logic should not belong to `CustomerIOClient` since it's `task-specific. How can we move it from her? if sub-tasks are bit overwork, maybe repositories/use-cases ?

Reply:
I am having a hard time understanding your question. What logic should not being in `CustomerIOClient` and how is that logic task specific? Being more specific about the code would be very helpful.

Reply:
Besides file size, do you see any problems with the `CustomerIOClient` as it is in this PR? 

For me, when tests are difficult to write, that's an indicator that a function or class is getting difficult to maintain. I only needed to write [5 test functions](https://github.com/customerio/customerio-android/blob/32096d119fd63dc17ca9d76ea5620452fe7d8d3f/sdk/src/sharedTest/java/io/customer/sdk/CustomerIOClientTest.kt#L59-L181=) for `identify` and I did not feel they were difficult to create. 

I like to avoid putting code into separate files to simply keep file sizes down. I do agree that large files are difficult to read and maintain, but I also find that lots of small files are also difficult to read and maintain. 

What do you think about breaking up `CustomerIOClient` into separate files based on their purpose? Some functions have to deal with profiles (identify), some have to do with tracking (screens, events, metrics), etc. Should we have a `IdentifiedProfileRepository`, `TrackingRepository` to break this file up?

Reply:
File size and somewhat a `chain-of-responsibility`, right now all of the logic is present in `CustomerIOClient` hence its responsible for storing/fetching identifiers, checking the blocking groups logic, etc. 

We should have a breakdown of this responsibility at some level, and I feel like repositories like you also mentioned could be a good first candidate for it.

Reply:
Since we want to get this feature released in 1-2 days, I [created a ticket](https://github.com/customerio/issues/issues/7160) to break up the `CustomerIOClient` file after this ships.